### PR TITLE
feat: [IE-109]: Single Build file to Build all service jars

### PR DESCRIPTION
**Pull request by Vbhatia20**
## Describe your changes
Single script file which takes care of building all services.
will remove Build script path as refactoring.
some extra manifest changes are also included to enable building of docker images and helm charts

Testing:
Tested and successfully build jars and docker images and feature helm chart for below services
platform-service
ng-manager 
ce-nextgen
pipeline-service 
migrator
change-data-capture
batch-processing
access-control
ci-manager
sto-manager
template-service 
cg-manager
Refer build execution below:
https://app.harness.io/ng/account/vpCkHKsDSxK9_KYfjCTMKA/ci/orgs/default/projects/FEATUREBUILDS/pipelines/MS_2/executions/6wQd32HxQfe5NZCElry9Nw/pipeline
Screenshot 2023-05-24 at 9 45 01 AM

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/48268)

### CODE REVIEW

**Comment by Vbhatia20** [comment]
trigger uts

File: build/feature_build.sh
Comment (diptiman-adak) [inline build/feature_build.sh:1]: Please move these files inside scripts folder

File: build/build_jar.sh [hunk 1]
```diff
@@ -0,0 +1,93 @@
+#!/usr/bin/env bash
+# Copyright 2021 Harness Inc. All rights reserved.
+# Use of this source code is governed by the PolyForm Free Trial 1.0.0 license
+# that can be found in the licenses directory at the root of this repository, also available at
+# https://polyformproject.org/wp-content/uploads/2020/05/PolyForm-Free-Trial-1.0.0.txt.
+
+set -ex
+
+if [ "${PLATFORM}" == "jenkins" ]; then
```
Comment (diptiman-adak) [inline build/build_jar.sh:9]: Let's remove this now, we are not using jenkins anymore

**Review by gn-harness — APPROVED** [review]
looks good. Just a few questions

1. where is CONTEXT_PATH used?
2. Shall the distribution creation functions such as copy_dms_jars live with the respective team's code?

File: 120-ng-manager/build/build_dist.sh [hunk 2]
```diff
@@ -32,9 +34,11 @@ cp ../../120-ng-manager/src/main/resources/enterprise-redisson-jcache.yaml .
 cp ../../120-ng-manager/src/main/resources/jfr/default.jfc .
 cp ../../120-ng-manager/src/main/resources/jfr/profile.jfc .
 
-cp ../../dockerization/ng-manager/Dockerfile-ng-manager-jenkins-k8-openjdk ./Dockerfile
+ls ../../dockerization
```
[the remark this answers is not in the document]
Comment (Vbhatia20) [reply 120-ng-manager/build/build_dist.sh:37]: done

File: 120-ng-manager/build/build_dist.sh [hunk 3]
```diff
@@ -10,7 +10,8 @@ set -ex
 
 mkdir -p dist/ng-manager
 cd dist/ng-manager
-
+pwd
+ls
```
[the remark this answers is not in the document]
Comment (Vbhatia20) [reply 120-ng-manager/build/build_dist.sh:14]: done

**Pull request closed without merging**